### PR TITLE
fix node_exists checks

### PR DESCRIPTION
**Pull request by piax93**
`jsoncfg` objects do not define a `node_exists` method, but they [override `__getattr__` with `__getitem__ `](https://github.com/pasztorpisti/json-cfg/blob/master/src/jsoncfg/config_classes.py#L285). This results in `self.policy_json.node_exists` not erroring out but returning a `ValueNotFoundNode` (because `jsoncfg` never raises key errors) and that for some reason [overrides `__call__`](https://github.com/pasztorpisti/json-cfg/blob/master/src/jsoncfg/config_classes.py#L155) with a method which ends up returning the first parameter passed.

TLDR, `self.policy_json.node_exists("Version")` would always return `"Version"`, hence passing the if-statement check. To fix this, it is simply enough to use the `jsoncfg.node_exists` utility function (which simply does a type check on the passed argument)

```
>>> import jsoncfg
>>> d = '{"foo": "bar"}'
>>> x = jsoncfg.loads_config(d)
>>> x.asd
<jsoncfg.config_classes.ValueNotFoundNode object at 0x10170ccd0>
>>> x.asd('123')
'123'
>>> jsoncfg.node_exists(x['foo'])
True
>>> jsoncfg.node_exists(x['none'])
False
```

### CODE REVIEW

**Comment by piax93** [comment]
@0xdabbad00 for review

**Comment by 0xdabbad00** [comment]
Woah good find! Thank you @piax93. LGTM. This can be merged @steiza
